File: src/main/java/miniservice/DeleteTableButton.java
package miniservice;

import dao.TableDefinitionDao;
import javafx.scene.control.Button;

public class DeleteTableButton extends Button {

	public DeleteTableButton(TableCountRecord record) {
		super("all delete");
		setOnMouseClicked(b -> {
			TableDefinitionDao dao;
			try {
				dao = new TableDefinitionDao();
				dao.deleteTable(record.tableNameProperty().get());
			} catch (Exception e) {
				// TODO 自動生成された catch ブロック
				e.printStackTrace();
			}
			record.countProperty().set(0);
		});

	}

}

File: src/main/java/tdc/DeleteTableButton.kt
package tdc

import javafx.event.EventHandler
import javafx.scene.control.Button
import javafx.scene.input.MouseEvent
import org.springframework.stereotype.Component
import tdc.dao.TableDefinitionDao

@Component
class DeleteTableButton(
    private val dao: TableDefinitionDao, private val record: TableCountRecord
) : Button("all delete") {
    init {
        onMouseClicked = EventHandler { b: MouseEvent? ->
            try {
                dao.deleteTable(record.tableNameProperty()!!.get())
            } catch (e: Exception) {
                // TODO 自動生成された catch ブロック
                e.printStackTrace()
            }
            record.countProperty()!!.set(0)
        }
    }
}